great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -503,49 +507,85 @@ class PandasDatasource(_PandasDatasource):
     def test_connection(self, test_assets: bool = True) -> None:
         ...
 
+    @staticmethod
+    def _validate_asset_name(asset_name: Optional[str] = None) -> str:
+        if asset_name == DEFAULT_PANDAS_DATA_ASSET_NAME:
+            raise PandasDatasourceError(
+                f"""An asset_name of {DEFAULT_PANDAS_DATA_ASSET_NAME} cannot be passed because it is a reserved name."""
+            )
+        if not asset_name:
+            asset_name = DEFAULT_PANDAS_DATA_ASSET_NAME
+        return asset_name
```

Comment:
If we do this as a standard pydantic validator method we will get the nice pydantic error locations/`loc` details

Reply:
I started to do that, but the issue is that we need to do type conversion `Optional[str]` -> `str`. There is no way to do type conversion in a validator and have `mypy` pick-up on the types correctly.

Reply:
@Kilo59 see: https://github.com/pydantic/pydantic/discussions/5050